gamestonk_terminal/portfolio/portfolio_view.py

```diff
@@ -198,38 +189,107 @@ def plot_rolling_beta(
     return ImageReader(imgdata)
 
 
-def annual_report(df: pd.DataFrame, hist: pd.DataFrame, m_tick: str) -> None:
-    """Generates an annual report
+def plot_ef(
+    stocks: List[str],
+    variance: float,
+    per_ret: float,
+    period: str = "3mo",
+    n_portfolios: int = 300,
+    risk_free: bool = False,
+):
+    """Display efficient frontier
 
     Parameters
     ----------
-    df : pd.DataFrame
-        The dataframe to be analyzed
-    m_tick : str
-        The ticker for the market asset
-    hist : pd.DataFrame
-        A dataframe of historical returns
+    stocks : List[str]
+        List of the stocks to be included in the weights
+    variance : float
+        The variance for the portfolio
+    per_ret : float
+        The portfolio's return for the portfolio
+    period : str
+        The period to track
+    n_portfolios : int
+        The number of portfolios to generate
+    risk_free : bool
+        Include the risk-free asset
     """
-    df_m = yfinance_model.get_market(df.index[0], m_tick)
-    dire = os.path.dirname(os.path.abspath(__file__)).replace(
-        "gamestonk_terminal", "exports"
+    fig, ax = plt.subplots(figsize=(10, 5), dpi=PLOT_DPI)
+    ef, rets, stds = optimizer_model.generate_random_portfolios(
+        [x.upper() for x in stocks], period, n_portfolios
     )
+    sharpes = rets / stds
+    ax.scatter(stds, rets, marker=".", c=sharpes, cmap="viridis_r")
+    plotting.plot_efficient_frontier(ef, ax=ax, show_assets=True)
+    # Find the tangency portfolio
+    rfrate = get_rf()
+    ret_sharpe, std_sharpe, _ = ef.portfolio_performance(risk_free_rate=rfrate)
+    ax.scatter(std_sharpe, ret_sharpe, marker="*", s=100, c="r", label="Max Sharpe")
+    plt.plot(variance, per_ret, "ro", label="Portfolio")
+    # Add risk free line
+    if risk_free:
+        y = ret_sharpe * 1.2
+        m = (ret_sharpe - rfrate) / std_sharpe
+        x2 = (y - rfrate) / m
+        x = [0, x2]
+        y = [rfrate, y]
+        line = Line2D(x, y, color="#FF0000", label="Capital Allocation Line")
+        ax.set_xlim(xmin=min(stds) * 0.8)
+        ax.add_line(line)
+    ax.set_title(f"Efficient Frontier simulating {n_portfolios} portfolios")
+    ax.legend()
+    fig.tight_layout()
+    ax.grid(b=True, which="major", color="#666666", linestyle="-")
+
+    if gtff.USE_ION:
+        plt.ion()
 
-    path = os.path.abspath(
-        os.path.join(
-            dire,
-            f"ar_{datetime.now().strftime('%Y%m%d_%H%M%S')}.pdf",
+    imgdata = BytesIO()
+    fig.savefig(imgdata, format="png")
+    imgdata.seek(0)
+    return ImageReader(imgdata)
+
+
+class Report:
+    def __init__(self, df: pd.DataFrame, hist: pd.DataFrame, m_tick: str, n: int):
+        self.df = df
+        self.hist = hist
+        self.m_tick = m_tick
+        self.df_m = yfinance_model.get_market(self.df.index[0], self.m_tick)
+        self.returns, self.variance = portfolio_model.get_return(df, self.df_m, n)
```

Comment:
You should add a docstring either to the class of to __init__ that explains what all the inputs are.  

Maybe not as intense, but check out https://github.com/scikit-learn/scikit-learn/blob/1f8825c8dd6238355191e3d1c98f4e4d54cfbf16/sklearn/linear_model/_bayes.py#L24. as an example